File: Range Sum Query - Immutable.cpp
class NumArray {
public:
vector<int>nums;
vector<int>prefix;

    NumArray(vector<int>& nums) {
        this->nums=nums;
        int n=nums.size();
        prefix.resize(n);
        prefixcal(nums);

    }
    
    int sumRange(int left, int right) {
        // return accumulate(this->nums.begin()+left,this->nums.begin()+right+1,0);
      return left-1>=0?prefix[right]-prefix[left-1]:prefix[right];  
    }
    void prefixcal(vector<int>nums){
    for(int i=0;i<nums.size();i++){
        (i==0)?prefix[i]=nums[i]:prefix[i]=prefix[i-1]+nums[i];
    }
    }
};

File: first-phase/week2/range-sum-query-immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.NumArray = nums
        self.prefix = []
        total = 0
        self.prefix.append(total)
        for i in nums:
            total += i
            self.prefix.append(total)
        print(self.prefix)

    def sumRange(self, left: int, right: int) -> int:
        return self.prefix[right+1] - self.prefix[left] 


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)
# 0 , 